Tensile/KernelWriterAssembly.py

```diff
@@ -3763,6 +3763,31 @@ def openPersistentLoop(self, kernel):
     if kernel["PersistentKernel"] or kernel["StreamK"]:
       if kernel["StreamK"]:
         # Workload calculations
+        if kernel["StreamKXCCMapping"]:
+          sXCC = self.sgprPool.checkOut(1, "XCC", preventOverflow=0)
+          sGridC = self.sgprPool.checkOut(1, "sGridC", preventOverflow=0)
+          sGridF = self.sgprPool.checkOut(1, "sGridF", preventOverflow=0)
+          sGridM = self.sgprPool.checkOut(1, "sGridM", preventOverflow=0)
+          kStr += inst("s_add_u32", sgpr(sGridC), sgpr("skGrid"), 7, "ceil(grid/8)")
```

Comment:
Does 8 mean number of XCD?

Reply:
Yes, I'm looking into a way of polling a Hip API for the number of XCCs to make this code more generic. Planning to work on that next week and post it in a future PR.

Reply:
Essentially, this seems to be similar to WorkGroupMapping.
We should be able to generalize this logic by using StreamKXCCMapping=number of XCD.

Reply:
But, we need inverse value for non power of 2 cases....
(WGM case, we get it from kernel parameter...)

Reply:
Technically, we have a macro function to calculated generic division (this is used in tensilelite), but I have never tried that before.

Reply:
At least, I would suggest to have an ability to support different number of XCD as a tensile config parameter.
Ideally, it is better generalize the code and use same kernel for any XCD numbers, but I do not have a good idea so far.

Reply:
I think what I said was not accurate.
If we know the divisor value when generating kernel, we can get magic divisor number and generate any division code.
Then, at least we can generate kernel code for any XCD numbers (we need to have different library for each XCD number)

For further generalization, we need to get number of XCD at runtime and pass the value through kernel parameter. Then, use generic division function to calculate this.
(we can refer division code from tensilelite WGM code)

Reply:
Yea I was hoping to make it poll for XCD count, but then it would also become a run-time parameter that needs extra kernel arguments, and the only gain would be consolidating kernels across a couple variants of gfx942 (which should have small number of kernels needed for stream-k anyway).
You're right, this is probably better as a tuning parameter that we can set. Then it's statically generated, doesn't require extra arguments, and we can also use it for different partitioning modes as well (when running on fewer than all XCDs on the device).
I'll work on refactoring the division this week.